src/Core/Traits/Events/DispatchesEvents.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * Add methods for dispatching events
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\Core\Traits\Events;
+
+use Illuminate\Contracts\Events\Dispatcher;
```

Comment:
So this creates an explicit dependency in our Core code on an external lib. It probably breaks some fancy SOLID rules but I couldn't bring myself to add more weird  misdirection to our code.